All work since UNH

Liammurray19 opened the pull request.

TimB-87 committed:
Auto, Vision, etc.

bot190 committed:
fix(drive): reduce measurement latency

Tune the Drive motor encoder settings to reduce the latency in the velocity measurements.

TimB-87 committed:
Fix shooter speeds and check-up autos

TimB-87 committed:
fix(camer): use correct camera angle

Update the camera translation to be the correct angle

TimB-87 committed:
fix(auto): resolve intake led issues

Move Grabbed Piece LED so that it runs after the intake sequence ends.

TimB-87 committed:
Update shooter stuff and autos

TimB-87 committed:
District updates

TimB-87 committed:
Added feeder shot

TimB-87 committed:
Autos and fix shooter speeds - DCMP

TimB-87 committed:
DCMP code - whoops

bot190 commented on src/main/java/frc/robot/subsystems/Led.java:
```diff
@@ -17,7 +17,7 @@ public class Led extends LEDSubsystem {
         super(map);
         // This one is length / 2 because the buffer has a mirrored other half
-        ledBuffer.setPattern("Fire", new ColdFirePattern(ledBuffer.getLength() / 2));
+        // ledBuffer.setPattern("Fire", new ColdFirePattern(ledBuffer.getLength() / 2));
```
Lets just get rid of this.

bot190 commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -142,24 +142,25 @@ public void configureButtonBindings() {
-        copilotController.b().whileTrue(commandSequences.scoreSpeakerCharge(copilotController, driveController));
-        copilotController.b().onFalse(commandSequences.scoreSpeakerRelease(copilotController, driveController));
-        copilotController.y().onTrue(commandSequences.scoreSpeakerCenterlineCharge(copilotController, driveController));
+        copilotController.b()
+                .whileTrue(commandSequences.charge(Speeds.SUBWOOFER_SHOT, ArmPresets.SCORE_SPEAKER_SUBWOOFER));
+        copilotController.b().onFalse(commandSequences.release());
```
```suggestion
        copilotController.b()
                .whileTrue(commandSequences.charge(Speeds.SUBWOOFER_SHOT, ArmPresets.SCORE_SPEAKER_SUBWOOFER)).onFalse(commandSequences.release());
```
These functions can be chained together instead of getting the button again.

bot190 replied:
And it looks like we're using `onTrue` instead of `whileTrue` for some of these below.

Liammurray19 committed:
condensed the shooter lines and got rid of unused led and shooter values

Liammurray19 committed:
added vision estimator

Liammurray19 committed:
takes vision measurements + logs it

Liammurray19 committed:
takes vision measurements + logs it

TimB-87 committed:
Rotate to angle - tuned and working

TimB-87 committed:
Merge branch 'UNH-work' of https://github.com/chopshop-166/frc-2024 into UNH-work

TimB-87 committed:
Robot knows where it is (by vision)

It's eyes have been opened, the fates have aligned

TimB-87 committed:
Get rid of legacy targeting

TimB-87 committed:
Robot can look at the speaker!

It's a tiny tiny bit off but we'll figure it out. Next step: drive integration.

bot190 commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -272,29 +304,26 @@ public void periodic() {
         estimator.update(getMap().gyro.getRotation2d(), getData().getModulePositions());
-        tgt = getSpeakerTarget();
+        visionEstimator.update(getMap().gyro.getRotation2d(), getData().getModulePositions());
+        Pose3d estimatorPose3D = new Pose3d(estimator.getEstimatedPosition());
```
Unused, remove

bot190 commented on src/main/java/frc/robot/subsystems/Shooter.java:
```diff
@@ -49,13 +49,17 @@ public Shooter(ShooterMap shooterMap) {
         PersistenceCheck speedPersistenceCheck = new PersistenceCheck(5, () -> {
-            return Math.abs(getData().leftWheels.velocity) > speed.getLeftSpeed(getMap().splitSpeeds)
-                    && Math.abs(getData().rightWheels.velocity) > speed.getRightSpeed(getMap().splitSpeeds);
+            return Math.abs(getData().leftWheels.velocity) > speed.getLeftSpeed(getMap().splitSpeeds) - 50
```
We should turn these numbers into constants `private final double BLAH_BLAH_BLAH`

bot190 commented on src/main/java/frc/robot/Robot.java:
```diff
@@ -141,28 +143,28 @@ public void configureButtonBindings() {
+        driveController.y().whileTrue(drive.rotateToSpeaker());
+        driveController.a().whileTrue(drive.rotateToAngle(90));
+        driveController.b().whileTrue(drive.rotateToAngle(180));
+        driveController.x().whileTrue(drive.rotateToAngle(270));
```
We can probably remove these now.

Liammurray19 committed:
got rid of un-used buttons and made bare numbers variables in shooter

TimB-87 committed:
Vision Testing - Make it rotate only the min distance it needs to

TimB-87 committed:
Merge branch 'UNH-work' of https://github.com/chopshop-166/frc-2024 into UNH-work

bot190 commented on src/main/java/frc/robot/subsystems/Drive.java:
```diff
@@ -304,8 +304,7 @@ public void periodic() {
         estimator.update(getMap().gyro.getRotation2d(), getData().getModulePositions());
-        visionEstimator.update(getMap().gyro.getRotation2d(), getData().getModulePositions());
-        Pose3d estimatorPose3D = new Pose3d(estimator.getEstimatedPosition());
+        visionEstimator.update(getMap().gyro.getRotation2d(), getData().getModulePositions());s
```
```suggestion
        visionEstimator.update(getMap().gyro.getRotation2d(), getData().getModulePositions());
```
Accidental typo! Don't forget to build before pushing.

TimB-87 committed:
Get rid of stupid code Liam did

TimB-87 committed:
Vision is precise

Doesn't point directly at target but points at one point consistently
